executor/src/witgen/machines/mod.rs

```diff
@@ -35,6 +36,21 @@ mod write_once_memory;
 /// A machine is a set of witness columns and identities where the columns
 /// are used on the right-hand-side of lookups. It can process lookups.
 pub trait Machine<'a, T: FieldElement>: Send + Sync {
+    /// Runs the machine without any arguments from the first row (.
+    fn run_timed<Q: QueryCallback<T>>(&mut self, mutable_state: &MutableState<'a, T, Q>) {
+        record_start(self.name());
+        self.run(mutable_state);
+        record_end(self.name());
+    }
+
+    /// Runs the machine without any arguments from the first row.
+    fn run<Q: QueryCallback<T>>(&mut self, _mutable_state: &MutableState<'a, T, Q>) {
+        unimplemented!(
+            "Running machine {} without a lookup is not supported.",
```

Comment:
```suggestion
            "Running machine {} without a machine call is not supported.",
```
Let's try to get away from calling machine connections lookups (or worse, plookups 😱)